DocsNav/scripts/insert_navbar.jl

```diff
@@ -0,0 +1,165 @@
+# insert_navbar.jl
+#
+# Usage:
+#   julia insert_navbar.jl <html-file-or-directory> <navbar-file-or-url> [--exclude "path1,path2,..."]
+#
+# Features:
+#   - Processes all .html files in the target (either a single file or recursively in a directory).
+#   - Removes any previously inserted navbar block (i.e. everything between <!-- NAVBAR START --> and <!-- NAVBAR END -->)
+#     along with any extra whitespace immediately following it.
+#   - Inserts the new navbar block immediately after the first <body> tag.
+#   - If the fetched navbar content does not contain the markers, it is wrapped (without further modification) with:
+#         <!-- NAVBAR START -->
+#         (navbar content)
+#         <!-- NAVBAR END -->
+#   - If a file does not contain a <body> tag, that file is skipped.
+#   - An optional --exclude parameter (comma‑separated) will skip any file whose path contains one of the provided substrings.
+#
+# Required package: HTTP
+#
+# (Install HTTP via Julia’s package manager, e.g. using Pkg; Pkg.add("HTTP"))
+
+using HTTP
+
+# --- Utility: Read file contents ---
+function read_file(filename::String)
+    open(filename, "r") do io
+        read(io, String)
+    end
+end
+
+# --- Utility: Write contents to a file ---
+function write_file(filename::String, contents::String)
+    open(filename, "w") do io
+        write(io, contents)
+    end
+end
+
+# --- Exclusion Function ---
+function should_exclude(filename::String, patterns::Vector{String})
+    for pat in patterns
+        if occursin(pat, filename)
+            return true
+        end
+    end
+    return false
+end
+
+# --- Remove any existing navbar block and any whitespace following it ---
+function remove_existing_navbar(html::String)
+    start_marker = "<!-- NAVBAR START -->"
+    end_marker   = "<!-- NAVBAR END -->"
+    while occursin(start_marker, html) && occursin(end_marker, html)
+        start_idx_range = findfirst(start_marker, html)
+        end_idx_range   = findfirst(end_marker, html)
+        # Extract the first index from the returned range.
+        start_idx = first(start_idx_range)
+        end_idx   = first(end_idx_range)
+        # Get prefix: everything before the start marker (or empty if start_idx is 1)
+        prefix = start_idx > 1 ? html[1:start_idx-1] : ""
```

Comment:
```suggestion
        # Get prefix: everything before the start marker (or empty if start_idx is 1)
        prefix = html[1:start_idx-1]
```

Indexing with an empty range works, so this could be simplified a bit:

```julia
julia> "hello"[1:0]
""
```